laurent22/joplin
# Desktop: Resolves #8369: Better describe decryption errors

# Summary
This **draft** pull request gives users better information about decryption errors on the desktop client. Notably, if 

# Testing plan
1. Create notes encrypted with an unknown encryption method on a sync target (TODO before this pull request is ready: Add an easy way to do this in dev mode)
2. Sync to this target
3. Click on the link to upgrade the client on the warning that appears
4. Click on the link to go to encryption settings on the warning
5. Click on the "check for updates" button in encryption settings
6. Navigate to the sync info screen from the decryption screen (using the "details" link)
7. Click "retry all" on the decryption issues listed on that screen.

# Screenshots
1. A new warning message is displayed just after decyption fails. Because the error message isn't stored, restarting the app displays a different message ("items could not be synced"), which also includes a link to the additional information page.
![](https://github.com/laurent22/joplin/assets/46334387/1a7a342f-ede2-4ef0-b6d3-bb7f8d77d473)

2. If items failed to decrypt, the encryption configuration page now links to the synchronization details page (which has a list of items that failed to decrypt). Additionally, it displays a message requesting the user to upgrade to a new version of Joplin if an unknown encryption method has been detected.
![](https://github.com/laurent22/joplin/assets/46334387/430b6f27-d7bf-4f58-ace1-c9f398ecaae9)

3. The synchronization report page now labels items that failed to decrypt with an error message, if an error message is available. Additionally, clicking "retry" now adds the item to the list again if it failed to decrypt again, rather than removing the item even if it failed to decrypt as a result of clicking "retry".
![](https://github.com/laurent22/joplin/assets/46334387/3337fe27-1447-4472-8da8-a375b06bbe26)


# To-do
- Add a set of `DecryptionWorker` tests that ensure it appropriately reports errors.
- Add a button or command (only in dev mode) to add a note encrypted with an unknown method to the database?

Resolves #8369
